### ISSUE TITLE
SVG image generated by \dot or \dotfile does not render properly in HTML

### ISSUE THREAD

**Issue by albert-github** [doxygen/doxygen#6869]
Getting the information from: 
doxygen assumes SVG size is specified in pt, what is not always the case (Origin: bugzilla #755484) #5876
as  supplied by @sawatts :

We are seeing these symptoms with current versions of Doxygen and GraphViz on RHEL7 x64 using either @dotfile or @dot. The input converts fine using dot directly, and we have not seen any issues with class or interaction diagrams generated by Doxygen from source.

Suspect that it is related to the interactive svg mode - adding the navigation controls to the top left. In this the link in the top right gives a "correct" graph. Unfortunately I don't want to lose this feature on the class diagrams etc.

I experimented with @dot , taking the first chunk of the imported dot file and adding nodes until the issue arose. This occurred after about 6 nodes in a flat (wide) structure. This wasn't sensitive to the content of the node or edge, but just the number (resultant width?).

I assume that Doxygen has some standard preamble for the dot graphs it generates. Since we control the external generation, it would be interesting to try using the same configuration?

Attached a minimal version showing the issue, including the output generated at our end ('html.tar' within this archive):

[issue-5876.tar.gz](https://github.com/doxygen/doxygen/files/2926449/issue-5876.tar.gz)

This consists of four cases - good & bad for dot and dotfile respectively. The bad cases are slightly larger graphs (one extra node) and trip the limit so that Doxygen adds the controls for interactive svg.

Diagrams generated by Doxygen itself do not suffer from this issue (and there are some quite large relationship diagrams generated in the real case!).

Since we control generation of the external dot files I can add properties to mirror what Doxygen might be using.

System Info:
```
> doxygen -v
1.8.15
> dot -V
dot - graphviz version 2.38.0 (20140413.2041)
> uname -a
Linux XXXXXXXXXXX 2.6.32-642.6.2.el6.x86_64 #1 SMP Mon Oct 24 10:22:33 EDT 2016 x86_64 x86_64 x86_64 GNU/Linux
> cat /etc/redhat-release
Red Hat Enterprise Linux Server release 6.8 (Santiago)
```

**Comment by albert-github** [comment]
I did a small test and to run the problem I had to adjust the Doxyfile, the DOT_PATH for me is on a different place.
I also saw CREATE_SUBDIRS set to YES and for small test cases I don't like to create all the sub directories, so I set it to NO and there was no problem, resetting it back to YES and the problem was back again.

Doxygen its "own" images don't "suffer" from this problem as these files are in the sub directories whilst the \dot and \dotfile images are in the root of the HTML output. In both cases the reference to svgpan.js is ../../svgpan.js but for \dot... files this should be just svgpan.js.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6870

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
